### ISSUE TITLE
upload-image doesn't finish

### ISSUE THREAD

**Issue by csharpknife2002** [kubevirt/kubevirt#3231]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What happened**:

I am using cdi along with virtctl to upload a window iso image via the following command:
kubectl-virt image-upload \
--image-path=/mnt/disks/images/chen-win-server-disk.qcow2 \
--pvc-name=win-svr-image \
--access-mode=ReadOnlyMany \
--pvc-size=25G \<br>
--uploadproxy-url=https://10.109.0.11:443 \
--insecure \
--wait-secs=1500

the file size is around 17G, after the upload finished I get the following log.

Uploading data completed successfully, waiting for processing to complete, you can hit ctrl-c without interrupting the progress. 

Then it stays there for even. ( I was able to upload two smaller images a few hours ago)

If I inspect the objects, I have the following result:
k get datavolume
NAME            PHASE         PROGRESS   AGE
iso-win2k12     Succeeded     N/A        3h51m
node-vm-disk    Succeeded     N/A        25h
win-svr-image   UploadReady   N/A        120m

k get pvc
NAME                    STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
iso-win2k12             Bound    pvc-f5a49723-e3e3-41cb-b960-50b9407758b5   5Gi        ROX            standard       3h53m
node-vm-disk            Bound    pvc-e2486ff6-5fbc-4f07-af66-ab694cadd491   25Gi       RWO            standard       25h
win-svr-image           Bound    pvc-1c911baa-bf4d-4589-9414-13e71a4b88b5   24Gi       ROX            standard       121m
win-svr-image-scratch   Bound    pvc-2d50fae6-d39f-4650-8159-6af1bde21d45   24Gi       RWO            standard       121m 

k get pods
NAME                              READY   STATUS    RESTARTS   AGE
cdi-upload-win-svr-image          1/1     Running   1          125m

k logs  cdi-upload-win-svr-image 
I0402 07:46:17.736610       1 uploadserver.go:62] Upload destination: /data/disk.img
I0402 07:46:17.736718       1 uploadserver.go:64] Running server on 0.0.0.0:8443

k logs -n kube-system kube-controller-manager-chen-k8s-master

I0402 07:29:46.846875       1 event.go:278] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"win-svr-image", UID:"1c911baa-bf4d-4589-9414-13e71a4b88b5", APIVersion:"v1", ResourceVersion:"471463", FieldPath:""}): type: 'Normal' reason: 'ProvisioningSucceeded' Successfully provisioned volume pvc-1c911baa-bf4d-4589-9414-13e71a4b88b5 using kubernetes.io/gce-pd
I0402 07:29:47.380831       1 event.go:278] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"win-svr-image-scratch", UID:"2d50fae6-d39f-4650-8159-6af1bde21d45", APIVersion:"v1", ResourceVersion:"471468", FieldPath:""}): type: 'Normal' reason: 'ProvisioningSucceeded' Successfully provisioned volume pvc-2d50fae6-d39f-4650-8159-6af1bde21d45 using kubernetes.io/gce-pd
I0402 07:29:48.652503       1 reconciler.go:275] attacherDetacher.AttachVolume started for volume "pvc-1c911baa-bf4d-4589-9414-13e71a4b88b5" (UniqueName: "kubernetes.io/gce-pd/kubernetes-dynamic-pvc-1c911baa-bf4d-4589-9414-13e71a4b88b5") from node "chen-k8s-node-2" 
I0402 07:29:48.652645       1 reconciler.go:275] attacherDetacher.AttachVolume started for volume "pvc-2d50fae6-d39f-4650-8159-6af1bde21d45" (UniqueName: "kubernetes.io/gce-pd/kubernetes-dynamic-pvc-2d50fae6-d39f-4650-8159-6af1bde21d45") from node "chen-k8s-node-2" 
I0402 07:29:53.349141       1 operation_generator.go:361] AttachVolume.Attach succeeded for volume "pvc-2d50fae6-d39f-4650-8159-6af1bde21d45" (UniqueName: "kubernetes.io/gce-pd/kubernetes-dynamic-pvc-2d50fae6-d39f-4650-8159-6af1bde21d45") from node "chen-k8s-node-2" 
I0402 07:29:53.349363       1 event.go:278] Event(v1.ObjectReference{Kind:"Pod", Namespace:"default", Name:"cdi-upload-win-svr-image", UID:"ea287917-80bb-4e7b-9e71-1f0e9c64c854", APIVersion:"v1", ResourceVersion:"471507", FieldPath:""}): type: 'Normal' reason: 'SuccessfulAttachVolume' AttachVolume.Attach succeeded for volume "pvc-2d50fae6-d39f-4650-8159-6af1bde21d45" 
I0402 07:29:56.987153       1 operation_generator.go:361] AttachVolume.Attach succeeded for volume "pvc-1c911baa-bf4d-4589-9414-13e71a4b88b5" (UniqueName: "kubernetes.io/gce-pd/kubernetes-dynamic-pvc-1c911baa-bf4d-4589-9414-13e71a4b88b5") from node "chen-k8s-node-2" 
I0402 07:29:56.987283       1 event.go:278] Event(v1.ObjectReference{Kind:"Pod", Namespace:"default", Name:"cdi-upload-win-svr-image", UID:"ea287917-80bb-4e7b-9e71-1f0e9c64c854", APIVersion:"v1", ResourceVersion:"471507", FieldPath:""}): type: 'Normal' reason: 'SuccessfulAttachVolume' AttachVolume.Attach succeeded for volume "pvc-1c911baa-bf4d-4589-9414-13e71a4b88b5" 

**What you expected to happen**:
The upload finish as expected

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`): 0.27.0
- Kubernetes version (use `kubectl version`): v1.18.0
- VM or VMI specifications: 
- Cloud provider or hardware configuration: gce
- OS (e.g. from /etc/os-release): debian 9
- Kernel (e.g. `uname -a`):
- Install tools: kubeadm
- Others:

et nodes -o wide
NAME              STATUS   ROLES    AGE   VERSION   INTERNAL-IP     EXTERNAL-IP     OS-IMAGE                       KERNEL-VERSION   CONTAINER-RUNTIME
chen-k8s-master   Ready    master   27h   v1.18.0   10.128.15.219      Debian GNU/Linux 9 (stretch)   4.9.0-12-amd64   docker://18.6.0
chen-k8s-node-1   Ready    <none>   26h   v1.18.0   10.128.15.231      Debian GNU/Linux 9 (stretch)   4.9.0-12-amd64   docker://18.6.0
chen-k8s-node-2   Ready    <none>   26h   v1.18.0   10.128.15.245      Debian GNU/Linux 9 (stretch)   4.9.0-12-amd64   docker://18.6.0

**Comment by DonSchenck** [comment]
I'm seeing the same issue.

**Comment by mhenriks** [comment]
Check the `cdi-upload-<pvc name>` pod logs.  If the pod restarted, check the previous log.  We've seen this happen when the target PVC needs more space.

**Comment by marccampbell** [comment]
thanks @mhenriks that unblocks me:

```
k logs -p cdi-upload-centos7-disk
I0603 15:56:23.666203       1 uploadserver.go:63] Upload destination: /data/disk.img
I0603 15:56:23.666251       1 uploadserver.go:65] Running server on 0.0.0.0:8443
I0603 15:56:30.914401       1 uploadserver.go:278] Content type header is ""
I0603 15:56:30.914428       1 data-processor.go:275] Calculating available size
I0603 15:56:30.915196       1 data-processor.go:283] Checking out file system volume size.
I0603 15:56:30.915212       1 data-processor.go:287] Request image size not empty.
I0603 15:56:30.915221       1 data-processor.go:292] Target size 2Gi.
I0603 15:56:30.915342       1 data-processor.go:205] New phase: TransferScratch
I0603 15:56:30.916048       1 util.go:144] Writing data...
I0603 15:56:34.230332       1 data-processor.go:205] New phase: Pause
I0603 15:56:34.230377       1 uploadserver.go:309] Returning success to caller, continue processing in background
I0603 15:56:34.230505       1 data-processor.go:150] Resuming processing at phase Process
I0603 15:56:34.230532       1 data-processor.go:205] New phase: Convert
I0603 15:56:34.230540       1 data-processor.go:211] Validating image
E0603 15:56:34.238436       1 data-processor.go:202] Virtual image size 8589934592 is larger than available size 2147483648. A larger PVC is required.
kubevirt.io/containerized-data-importer/pkg/image.(*qemuOperations).Validate
        pkg/image/qemu.go:193
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).validate
        pkg/importer/data-processor.go:212
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
        pkg/importer/data-processor.go:221
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
        pkg/importer/data-processor.go:189
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataResume
        pkg/importer/data-processor.go:152
kubevirt.io/containerized-data-importer/pkg/uploadserver.(*uploadServerApp).uploadHandlerAsync.func1
        pkg/uploadserver/uploadserver.go:299
runtime.goexit
        src/runtime/asm_amd64.s:1357
Image validation failed
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).validate
        pkg/importer/data-processor.go:214
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).convert
        pkg/importer/data-processor.go:221
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
        pkg/importer/data-processor.go:189
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataResume
        pkg/importer/data-processor.go:152
kubevirt.io/containerized-data-importer/pkg/uploadserver.(*uploadServerApp).uploadHandlerAsync.func1
        pkg/uploadserver/uploadserver.go:299
runtime.goexit
        src/runtime/asm_amd64.s:1357
Unable to convert source data to target format
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataWithPause
        pkg/importer/data-processor.go:191
kubevirt.io/containerized-data-importer/pkg/importer.(*DataProcessor).ProcessDataResume
        pkg/importer/data-processor.go:152
kubevirt.io/containerized-data-importer/pkg/uploadserver.(*uploadServerApp).uploadHandlerAsync.func1
        pkg/uploadserver/uploadserver.go:299
runtime.goexit
        src/runtime/asm_amd64.s:1357
E0603 15:56:34.238555       1 uploadserver.go:300] Error during resumed processing: Unable to convert source data to target format: Image validation failed: Virtual image size 8589934592 is larger than available size 2147483648. A larger PVC is required.
I0603 15:56:34.238570       1 uploadserver.go:307] Wrote data to /data/disk.img
E0603 15:56:34.238598       1 uploadserver.go:153] HTTP server returned error Unable to convert source data to target format: Image validation failed: Virtual image size 8589934592 is larger than available size 2147483648. A larger PVC is required.
E0603 15:56:34.238617       1 uploadserver.go:69] UploadServer failed: Unable to convert source data to target format: Image validation failed: Virtual image size 8589934592 is larger than available size 2147483648. A larger PVC is required.
```

**Comment by tylerauerbeck** [comment]
@mhenriks Just a +1 as this was my issue as well. Any idea if there are plans of better ways to bubble this error up or kill the upload right away if the PVC isn't large enough?
